Emby.Server.Implementations/Data/SqliteItemRepository.cs

```diff
@@ -4593,10 +4593,19 @@ private List<string> GetWhereClauses(InternalItemsQuery query, IStatement statem
 
             if (query.ExcludeInheritedTags.Length > 0)
             {
-                var tagValues = query.ExcludeInheritedTags.Select(i => "'" + GetCleanValue(i) + "'");
-                var tagValuesList = string.Join(",", tagValues);
-
-                whereClauses.Add("((select CleanValue from itemvalues where ItemId=Guid and Type=6 and cleanvalue in (" + tagValuesList + ")) is null)");
+                var paramName = "@ExcludeInheritedTags";
+                if (statement == null)
+                {
+                    int index = 0;
+                    whereClauses.Add("((select CleanValue from itemvalues where ItemId=Guid and Type=6 and cleanvalue in (" + string.Join(",", query.ExcludeInheritedTags.Select(t => paramName + index++)) + ")) is null)");
```

Comment:
Could you break this up into multiple statements? It's a bit hard to follow just glancing at it.

Reply:
No problem - I've split this back out into a separate for loop to build up the list of parameters.  I'd just in-lined it to avoid creating a new `List<string>`.

Reply:
Could also have been
```CS
string excludedTags = string.Join(",", query.ExcludeInheritedTags.Select(t => paramName + index++));
whereClauses.Add("((select CleanValue from itemvalues where ItemId=Guid and Type=6 and cleanvalue in (" + excludedTags + ")) is null)");
```

Reply:
Use an array instead of a list, you know the size in advance

Reply:
I wasn't sure that just using the same functionality to generate it into a separate string was enough.  :)

Which way would be preferable and I can update?

Reply:
Between these 3 options: LINQ, for loop + array or a StringBuilder, I would prefer the StringBuilder

Reply:
Pushed that change.